msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,161 @@
+          "supportedObjects": {
+            "file": {
+              "extensions": ["xlsx", "doc", "docx", "pdf", "pptx", "ppt"]
+            }
+          },
+          "pageInfo": {
+            "pageId": "index",
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+For more information, see [public developer preview app manifest schema](../resources/schema/manifest-schema-dev-preview.md#actions).
+
+When a user selects an action to open a personal tab and view related tasks in an app based on the selected file. The app uses the `"intent": "custom"` property to identify the file type, such as .xlsx or doc and the `"type": "openPage"` handler opens the app and navigates to the `pageId`.
+
+## Retrieve Action information through context object
+
+Build the handler to receive the Action information through the [context object](/javascript/api/%40microsoft/teams-js/app.context?view=msteams-client-js-latest&preserve-view=true) to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
+
+When a user selects Add option from the app's context menu, a personal tab opens with the help of the `openPage` property in the app manifest. Your app can access contextual information about the invoked Action from the `actionInfo` property of the `app.getContext()` context object.
+
+The [ActionInfo](/javascript/api/@microsoft/teams-js/actioninfo) interface helps to enable your app to determine when a user opens a page from an Action and the content that initiated the Action.
+
+```javascript
+async setItemId() {
+    const context = await app.getContext();
+    const itemId = context.actionInfo && context.actionInfo.actionObjects[0].itemId;
+    this.setState({
+        itemId: itemId
+    });
+}
+```
+
+| &nbsp; | Name | Description |
+| --- | --- | --- |
+| &nbsp; | `actionObjects` | Array of corresponding action objects. |
+| &nbsp; | `itemId` | The app receives the ID as the content and uses it to query the Microsoft Graph. |
+
+## Access content through Graph API
+
+After obtaining the `itemId` of the triggering content, you can use the Graph API to read or modify the content, facilitating task completion for your users.
+
+```javascript
+app.getContext().then((context: app.Context) => {
+    const actionInfo = context.actionInfo;
+    if (actionInfo) {
+        // App was launched using an action    
+    } 
+    if (actionInfo && actionInfo.actionId == 'myActionId1') {
+        // Handle specific action    
+    } 
+    if (actionInfo) {
+        if (actionInfo.actionObject.type == app.ActionObjectType.M365Content) {
+            const itemId = actionInfo.actionObjects[0].itemId;
+            // Get the requested content from Mirosoft Graph by item id:
+        } 
+    }
+})
+```
+
+### Enable Actions in Microsoft Admin Center
+
+> [!NOTE]
+> Actions are available in [public developer preview](../resources/dev-preview/developer-preview-intro.md).
```

Comment:
This note is not saying anything. If you want to emphasize the public preview. You may say. 

Note:
Actions are in public preview, please upload the app package with devPreview manifest. And enroll your users into Targeted Release to experience it.